### PR TITLE
Enable 4 condition lint rules

### PR DESCRIPTION
**Pull request by fregante**
- Based on https://github.com/pixiebrix/pixiebrix-extension/pull/761

It's best to merge that one first.

This is essentially a whitespace-only lint change around else/if/return, no functional changes were made to the code. It felt like an easy fix to reduce indentation.

**PR diff best looked at without whitespace changes.**

- https://eslint.org/docs/rules/no-else-return <-- most common
- https://eslint.org/docs/rules/no-useless-return
- https://eslint.org/docs/rules/no-lonely-if
- https://eslint.org/docs/rules/no-negated-condition (not autofixable, so I just soft-enabled it)

In short, it turns else/if cascades:

```js
if (something) {
	return 1;
} else if (somethingElse) {
	return 2;
}
```

into in early returns:

```js
if (something) {
	return 1;
}

if (somethingElse) {
	return 2;
}
```

### CODE REVIEW

**Review by twschiller — APPROVED** [review]

**Comment by twschiller** [comment]
@fregante go ahead and merge once you've merged the other PR

image

**Comment by fregante** [comment]
These rules are the default in https://github.com/xojs/eslint-config-xo, which is a stricter preset than `eslint:recommended`. Eventually the explicit rules will be dropped from the config and replaced with the preset itself.

I looked into enabling that preset but instead of spending too much time linting the codebase now I'll probably apply some rules over time, like these.

These are the errors that it found on the codebase. Some are pretty easy and obvious, others need to be double-checked:

 309 [padding-line-between-statements](https://eslint.org/docs/rules/padding-line-between-statements)
 213 [capitalized-comments](https://eslint.org/docs/rules/capitalized-comments)
 136 [arrow-body-style](https://eslint.org/docs/rules/arrow-body-style)
  40 [no-void](https://eslint.org/docs/rules/no-void)
  32 [spaced-comment](https://eslint.org/docs/rules/spaced-comment)
  32 [no-warning-comments](https://eslint.org/docs/rules/no-warning-comments)
  21 [object-shorthand](https://eslint.org/docs/rules/object-shorthand)
  21 [no-implicit-coercion](https://eslint.org/docs/rules/no-implicit-coercion)
  19 [dot-notation](https://eslint.org/docs/rules/dot-notation)
  18 [prefer-destructuring](https://eslint.org/docs/rules/prefer-destructuring)
  13 [prefer-arrow-callback](https://eslint.org/docs/rules/prefer-arrow-callback)
   6 [no-return-await](https://eslint.org/docs/rules/no-return-await)
   6 [no-promise-executor-return](https://eslint.org/docs/rules/no-promise-executor-return)
   5 [no-useless-constructor](https://eslint.org/docs/rules/no-useless-constructor)
   4 [max-depth](https://eslint.org/docs/rules/max-depth)
   4 [func-call-spacing](https://eslint.org/docs/rules/func-call-spacing)
   4 [complexity](https://eslint.org/docs/rules/complexity)
   3 [no-useless-concat](https://eslint.org/docs/rules/no-useless-concat)
   3 [new-cap](https://eslint.org/docs/rules/new-cap)
   2 [no-self-compare](https://eslint.org/docs/rules/no-self-compare)
   2 [no-alert](https://eslint.org/docs/rules/no-alert)
   2 [array-callback-return](https://eslint.org/docs/rules/array-callback-return)
   2 [array-bracket-newline](https://eslint.org/docs/rules/array-bracket-newline)
   1 [radix](https://eslint.org/docs/rules/radix)
   1 [prefer-promise-reject-errors](https://eslint.org/docs/rules/prefer-promise-reject-errors)
   1 [prefer-object-spread](https://eslint.org/docs/rules/prefer-object-spread)
   1 [one-var](https://eslint.org/docs/rules/one-var)
   1 [no-unmodified-loop-condition](https://eslint.org/docs/rules/no-unmodified-loop-condition)
   1 [max-params](https://eslint.org/docs/rules/max-params)
   1 [lines-between-class-members](https://eslint.org/docs/rules/lines-between-class-members)
   1 [guard-for-in](https://eslint.org/docs/rules/guard-for-in)
   1 [func-names](https://eslint.org/docs/rules/func-names)
   1 [default-param-last](https://eslint.org/docs/rules/default-param-last)
